app/controllers/api/auth_key_pairs_controller.rb

```diff
@@ -1,5 +1,13 @@
 module Api
   class AuthKeyPairsController < BaseController
+    def create_resource(_type, _id = nil, data = {})
+      ext_management_system = ExtManagementSystem.find(data['ems_id'])
+      task_id = ManageIQ::Providers::CloudManager::AuthKeyPair.create_key_pair_queue(session[:userid], ext_management_system, data)
```

Comment:
We should probably check `.validate_create_key_pair` before calling `create_`

Reply:
Well, in that case I will need a way to determine the namespaced subclass of the `AuthKeyPair` under a given ems ...

Reply:
Isn't that what `class_by_ems` does?  e.g. https://github.com/ManageIQ/manageiq/blob/master/app/models/manageiq/providers/cloud_manager/auth_key_pair.rb#L41